sql/core/src/test/scala/org/apache/spark/sql/MathExpressionsSuite.scala

```diff
@@ -183,6 +184,18 @@ class MathExpressionsSuite extends QueryTest {
     testOneToOneMathFunction(floor, math.floor)
   }
 
+  test("factorial") {
+    val df = (0 to 5).map(i => (i, i)).toDF("a", "b")
+    checkAnswer(
+      df.select(factorial('a)),
+      (0 to 5).map(n => Row(IntMath.factorial(n)))
```

Comment:
Give the actual value(rows) for the expected results, which probably make more sense for reading the code.